connect/runtime/src/main/java/org/apache/kafka/connect/util/PluginVersionUtils.java

```diff
@@ -0,0 +1,186 @@
+package org.apache.kafka.connect.util;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.Connector;
+import org.apache.kafka.connect.runtime.ConnectorConfig;
+import org.apache.kafka.connect.runtime.isolation.PluginDesc;
+import org.apache.kafka.connect.runtime.isolation.Plugins;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorType;
+import org.apache.maven.artifact.versioning.InvalidVersionSpecificationException;
+import org.apache.maven.artifact.versioning.VersionRange;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class PluginVersionUtils {
+
+    private static Plugins plugins = null;
+
+    public static void setPlugins(Plugins plugins) {
+        PluginVersionUtils.plugins = plugins;
+    }
+
+    public static VersionRange connectorVersionRequirement(String version) throws InvalidVersionSpecificationException {
+        if (version == null || version.equals("latest")) {
+            return null;
+        }
+        version = version.trim();
+
+        // check first if the given version is valid
+        VersionRange.createFromVersionSpec(version);
+
+        // now if the version is not enclosed we consider it as a hard requirement and enclose it in []
+        if (!version.startsWith("[") && !version.startsWith("(")) {
+            version = "[" + version + "]";
+        }
```

Comment:
I think a more robust check for this would be to check if the recommendedVersion is non-null. (or i think equivalently, hasRestrictions)